baselines/diabetic_retinopathy_detection/data_load_scripts/load_aptos.py

```diff
@@ -0,0 +1,46 @@
+"""
```

Comment:
I think these 3 files can be simplified into one. I'm giving the gist of it here.

Given the use-case, I'd suggest "modes" for the datasets, ie one flag value dictates the other ones vs passing everything.

nit: If I understand this code correctly, it's not really loading the data vs reconfiguring it into a different structure and writing it. I propose renaming the script to something that reflects that. "prepare_dataset.py" could be sufficient. Also if we go with a single file, you could consider not creating the data_load_scripts subdirectory. 

# I noticed that download_dir is often but not always dataset_name. Defer to you if that's accurate and both args are actually needed.
def _download_and_prepare_dataset(dataset_name: str, split: str, data_dir: str, download_dir: str) -> None:
  builder = ub.datasets.get(dataset_name=dataset_name, split=split,data_dir=data_dir,download_data=True)
  builder._dataset_builder.download_and_prepare(download_dir=f'{data_dir}/{download_dir}/')

# A flag with a fixed value set, maybe using a validator: https://github.com/google/uncertainty-baselines/blob/89dff01c9492f51d5283d92ebce4d99616032a46/baselines/imagenet/hetsngp.py#L83

# Supported datasets. Either use those long strings as the flag values or come up with shorter ones to signify each "mode"
_APTOS = "aptos"
_DIABETIC_RETINOPATHY_SEVERITY_SHIFT_MILD = "diabetic_retinopathy_severity_shift_mild"
_UB_DIABETIC_RETINOPATH_DETECTION ="ub_diabetic_retinopathy_detection"

_SPLITS = {
_APTOS = [ < str for each split],
_DIABETIC_RETINOPATHY_SEVERITY_SHIFT_MILD = [ < str for each split],
...
}

# See comment above about whether we need this.
_DOWNLOAD_DIR = {
_APTOS = <value>
_DIABETIC_RETINOPATHY_SEVERITY_SHIFT_MILD = <value>
...
}


in the main:
  dataset_name = <from flag directly, or a dict lookup using the flag>
  for split in _SPLITS[dataset_name]:
     _download_and_prepare_dataset(dataset_name=dataset_name, split=split, data_dir=FLAGS.data_dir, download_dir=_DOWNLOAD_DIR[dataset_name])
     logging.info(f"Finished packaging {dataset_name} {split} split data.")

Please let me know if I missed something.

Reply:
Thanks for the thorough explanation + review! This will definitely help streamline the download/setup process for RETINA. I really appreciate it!